docs/usage_guide/optimization_guide/index.rst

```diff
@@ -0,0 +1,58 @@
+Optimization Guide
+==================
+
+Here you can find information on Modin optimizations both for general query pipeline and for specific operations.
+
+Generic settings
+""""""""""""""""
+
+If you are working with a wide DataFrame and performing an operation that requires ``map`` Core Dataframe Algebra operator
+to be applied cell-wise, you may want to change the number of partitions DataFrame should partition into to avoid
+excessive amount of remote partitions, which is much greater than the number of CPUs.
+
+.. code-block:: python
+
+  from multiprocessing import cpu_count
+  import modin.config as cfg
+  import modin.pandas as pd
+  import numpy as np
+
+  data = np.random.randint(0, 100, size=(2**15, 2**15))
+
+  cfg.NPartitions.put(cpu_count())
+  df = pd.DataFrame(data)
+  %timeit df.abs()
+  1.9 s  49.8 ms per loop (mean  std. dev. of 7 runs, 1 loop each)
+
+  cfg.NPartitions.put(int(cpu_count() / 2))
+  df = pd.DataFrame(data)
+  %timeit df.abs()
+  506 ms  17.5 ms per loop (mean  std. dev. of 7 runs, 1 loop each)
+
+Operation-specific settings
+"""""""""""""""""""""""""""
+
+merge
+-----
+
+``merge`` operation in Modin uses broadcast join algorithm: combining a right Modin DataFrame into pandas DataFrame and
+broadcasting it to row partitions of the left Modin DataFrame. In order to minimize interprocess communication cost when doing
+inner join you may want to swap left and right DataFrames.
```

Comment:
It's actually strange, that the user should worry about this, I think we have to treat it as an issue and fix it on the Modin side.